include/swift/SIL/SILModule.h

```diff
@@ -163,6 +163,8 @@ class SILModule {
       llvm::ilist<SILDifferentiabilityWitness>;
   using CoverageMapCollectionType =
       llvm::MapVector<StringRef, SILCoverageMap *>;
+  using BasicBlockNameMapType =
+      llvm::DenseMap<const SILBasicBlock *, llvm::StringRef>;
```

Comment:
From the `StringRef` comments
```
  /// This class does not own the string data, it is expected to be used in
  /// situations where the character data resides in some other buffer, whose
  /// lifetime extends past that of the StringRef. For this reason, it is not in
  /// general safe to store a StringRef.
```

I expected the stored type to be a `std::string`. I think if you just change the type in the map everything will just work, and the API's can still accept and vend `StringRef`, which is suitable for printing.

Reply:
Oh, didn't realize that -- some other collections did store StringRef here so I followed that.

Moving to std::string then 👍